### PR TITLE
Heatmap Graph Integration

### PR DESCRIPTION
**Pull request by nxjiten**
@natabal 

Convert heatmap graph from class to functional component. Please have a look.

heatmapgraph

### CODE REVIEW

File: GraphsV2/Heatmap/default.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export const properties = {
+    stroke: {
+        color: "grey",
+        width: "0.5px"
+    },
+    legend: {
+        show: false
+    },
+    mapColors: {
+        'InSla': theme.palette.greenColor,
```
Comment (natabal) [inline GraphsV2/Heatmap/default.config.js]: these colors should not be defined here, they are very specific to the application.

File: GraphsV2/Heatmap/index.js [hunk 2]
```diff
@@ -0,0 +1,738 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from "react"
+import isEmpty from 'lodash/isEmpty';
+import isEqual from 'lodash/isEqual';
+import * as d3 from "d3"
```
Comment (natabal) [inline GraphsV2/Heatmap/index.js]: please import only what you are using from d3

File: GraphsV2/Heatmap/heatmap.spec.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            expect(height).toEqual(1);
+        });
+
+        it("Legends", () => {
+            const legend = $('.legend').children().length;
+            expect(legend).toEqual(2);
+        });
+    });
+
+});
```
Comment (nxanil) [inline GraphsV2/Heatmap/heatmap.spec.js]: Add a new line please.

File: GraphsV2/Heatmap/mock/data/data.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    "ReachabilityStatus": 1
+  },
+  {
+    "key_as_string": "true",
+    "date_histo": 1584559800000,
+    "doc_count": 2,
+    "dnsg": "ovs-3",
+    "ReachabilityStatus": 1
+  }
+]
```
Comment (nxanil) [inline GraphsV2/Heatmap/mock/data/data.json]: Add a new line.

**Pull request merged**